TrafficCapture/dockerSolution/src/main/docker/migrationConsole/osiMigration.py

```diff
@@ -0,0 +1,385 @@
+        secret_config = secret_config.replace(AWS_REGION_PLACEHOLDER, aws_region)
+        pipeline_config = pipeline_config.replace(AWS_SECRET_CONFIG_PLACEHOLDER, secret_config)
+        pipeline_config = pipeline_config.replace(SOURCE_AUTH_OPTIONS_PLACEHOLDER, SOURCE_BASIC_AUTH_CONFIG_TEMPLATE)
+    else:
+        pipeline_config = pipeline_config.replace(AWS_SECRET_CONFIG_PLACEHOLDER, "")
+
+    if source_auth_type == 'SIGV4':
+        aws_source_config = SIGV4_AUTH_CONFIG_TEMPLATE.replace(AWS_REGION_PLACEHOLDER, aws_region)
+        aws_source_config = aws_source_config.replace(STS_ROLE_ARN_PLACEHOLDER, source_pipeline_role_arn)
+        pipeline_config = pipeline_config.replace(SOURCE_AUTH_OPTIONS_PLACEHOLDER, aws_source_config)
+
+    if target_auth_type == 'SIGV4':
+        aws_target_config = SIGV4_AUTH_CONFIG_TEMPLATE.replace(AWS_REGION_PLACEHOLDER, aws_region)
+        aws_target_config = aws_target_config.replace(STS_ROLE_ARN_PLACEHOLDER, target_pipeline_role_arn)
+        pipeline_config = pipeline_config.replace(TARGET_AUTH_OPTIONS_PLACEHOLDER, aws_target_config)
+
+    pipeline_config = pipeline_config.replace(SOURCE_ENDPOINT_PLACEHOLDER, source_endpoint)
+    pipeline_config = pipeline_config.replace(TARGET_ENDPOINT_PLACEHOLDER, target_endpoint)
+    return pipeline_config
+
+
+def start_pipeline(osi_client, pipeline_name: str):
+    logger.info(f"Starting pipeline: {pipeline_name}")
+    osi_client.start_pipeline(
+        PipelineName=pipeline_name
+    )
+
+
+def stop_pipeline(osi_client, pipeline_name: str):
+    logger.info(f"Stopping pipeline: {pipeline_name}")
+    osi_client.stop_pipeline(
+        PipelineName=pipeline_name
+    )
+
+
+def osi_create_pipeline(osi_client, pipeline_name: str, pipeline_config: str, subnet_ids: List[str],
+                        security_group_ids: List[str], cw_log_group_name: str, tags: List[Dict[str, str]]):
+    logger.info(f"Creating pipeline: {pipeline_name}")
+    cw_log_options = {
+        'IsLoggingEnabled': False
+    }
+    # Seeing issues from CW log group naming that need further investigation
+    # if cw_log_group_name is not None:
+    #     cw_log_options = {
+    #         'IsLoggingEnabled': True,
+    #         'CloudWatchLogDestination': {
+    #             'LogGroup': cw_log_group_name
+    #         }
+    #     }
+
+    osi_client.create_pipeline(
+        PipelineName=pipeline_name,
+        MinUnits=2,
+        MaxUnits=4,
+        PipelineConfigurationBody=pipeline_config,
+        LogPublishingOptions=cw_log_options,
+        VpcOptions={
+            'SubnetIds': subnet_ids,
+            'SecurityGroupIds': security_group_ids
+        },
+        # Documentation lists this as a requirement but does not seem to be the case from testing:
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/osis/client/create_pipeline.html
+        # BufferOptions={
+        #    'PersistentBufferEnabled': False
```

Comment:
I think we want false since we are a pull based source, but I experienced an issue with deployment essentially saying BufferOptions shouldn't exist, yet its a requirement on the API doc 🤷‍♂️